### DISCUSSION TITLE
Set terminal_background only in vscode

### DISCUSSION THREAD

**Discussion by floh96** [JanDeDobbeleer/oh-my-posh#605] [Troubleshoot] [answered]
Hi, 
is it possible to set the terminal_background setting only if the terminal is vscode?
vscode sets the env variable TERM_PROGRAM to vscode.
Is it possible to configure this in the theme.json to only apply the terminal_background setting if the env is matching?
Thank you for your answers :)

**Comment by JanDeDobbeleer** [comment, marked as answer]
Can do. I will add that 👍🏻

**Reply by JanDeDobbeleer** [reply]
@floh96 if I'm not mistaken we only made this feature FOR vscode, right? I suppose you're not using it in Windows Terminal?

**Reply by floh96** [reply]
@JanDeDobbeleer see https://github.com/JanDeDobbeleer/oh-my-posh/discussions/182#discussioncomment-243983 the problem seems to also occur when you are using the acrylic effect in windows terminal. I'm personally not using that at the moment so I'm also not using the terminal_background setting in windows terminal.

**Reply by JanDeDobbeleer** [reply]
In that case we'll need an array of terminal names to ensure we're allowing people to overrule this.

**Reply by JanDeDobbeleer** [reply]
@floh96 I'll add template functionality to that property meaning you can add this: `{{ if eq \"vscode\" .Env.TERM_PROGRAM }}#123456{{end}}`

**Reply by floh96** [reply]
@JanDeDobbeleer thank you 👍 with the template functionality is it also possible to only apply the setting on windows? (because the bug does only affect windows, see your bug report for reference https://github.com/microsoft/vscode/issues/111762).
Edit: then it would also be needed for wsl i think since this is also running on windows... so things would get complicated pretty quickly so i just enable it on all operating systems.

**Reply by JanDeDobbeleer** [reply]
You can use the env vars to create complicated logic using the templating engine.

**Reply by floh96** [reply]
@JanDeDobbeleer it does work thank you 👍 but now my vscode updated to 1.55.0 and now the prompt looks broken see screenshot 😞 is there something i can do to prevent this?
![Unbenannt](https://user-images.githubusercontent.com/49693964/113352413-82fef580-933c-11eb-9ac0-9d0de97b5369.PNG)

**Reply by JanDeDobbeleer** [reply]
Hmm. That's a font setting on their end. It's either looking at changing the font size or creating an issue because of regression. I know they have multiple engines, maybe they enabled a nee one by default.

**Reply by floh96** [reply]
@JanDeDobbeleer they changed the renderer see https://github.com/microsoft/vscode/issues/120371 and https://github.com/microsoft/vscode/issues/120129. I will test tomorrow if I can use the old one. 
Edit: it was an upstream issue in xtermjs which has been fixed in https://github.com/xtermjs/xterm.js/pull/3279 and was released to vscode insiders. 🎉 so I'm waiting for a new vscode release.
![vscode_insiders_fixed](https://user-images.githubusercontent.com/49693964/113411464-19bbc880-93b6-11eb-85cb-3cebfe5d36bf.PNG)
